crates/torii/graphql/src/tests/subscription_test.rs

```diff
@@ -101,10 +100,10 @@ mod tests {
     }
 
     #[sqlx::test(migrations = "../migrations")]
+    #[serial]
     async fn test_component_subscription(pool: SqlitePool) {
         // Sleep in order to run this test at the end in a single thread
         tokio::time::sleep(Duration::from_secs(2)).await;
```

Comment:
Since there is serial now, can we remove this sleep here as well?

Reply:
Yes, Before we removed that sleep(), but then we realized 'cargo test' failed.
 So we've added #[serial] to entities_test and others test as well as we detected sometimes these tests were executed before the subscription ones making them fail. A tokio::semaphore has been added to ensure that the task locks the thread completely and unlocking it inside the thread that publish the object to the subscription in that way all test pass secuentially